Tasks/Maven/mavencodeanalysis.ts

```diff
@@ -0,0 +1,231 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../definitions/Q.d.ts" />
+
+import path = require('path');
+import fs = require('fs');
+
+import tl = require('vsts-task-lib/task');
+import trm = require('vsts-task-lib/toolrunner');
+
+// Lowercased names are to lessen the likelihood of xplat issues
+import pmd = require('./pmdformaven');
+import ar = require('./analysisresult');
+import ma = require('./moduleanalysis');
+
+// Set up for localization
+tl.setResourcePath(path.join( __dirname, 'task.json'));
+
+// Cache build variables - if they are null, we are in a test env and can use test inputs
+// The artifact staging directory will be a subdirectory just to be safe.
+var sourcesDir:string = tl.getVariable('build.sourcesDirectory') || tl.getInput('test.sourcesDirectory');
```

Comment:
Ah. Yes, these strings (the tl.getInput() variable names) are duplicated in the tests, but (as in another discussion) since we can't reference this file directly anyway, it's a slightly moot point.